src/stacked_borrows/mod.rs

```diff
@@ -747,15 +697,15 @@ trait EvalContextPrivExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         place: &MPlaceTy<'tcx, Provenance>,
         size: Size,
         kind: RefKind,
+        retag_cause: RetagCause, // What caused this retag, for diagnostics only
         new_tag: SbTag,
         protect: bool,
     ) -> InterpResult<'tcx, Option<AllocId>> {
         let this = self.eval_context_mut();
-        let current_span = &mut this.machine.current_span();
 
         // It is crucial that this gets called on all code paths, to ensure we track tag creation.
         let log_creation = |this: &MiriEvalContext<'mir, 'tcx>,
-                            current_span: &mut CurrentSpan<'_, 'mir, 'tcx>,
+                            current_span: CurrentSpan<'_, 'mir, 'tcx>, threads: &'_ ThreadManager<'mir, 'tcx>,
```

Comment:
only one argument per line please

(rustfmt seems to just give up on this code)

Reply:
It sure does. It's pretty annoying.